docs/src/main/asciidoc/se/guides/upgrade_4x.adoc

```diff
@@ -72,6 +72,25 @@ include::{sourcedir}/se/guides/Upgrade4xSnippets.java[tag=snippet_2, indent=0]
 
 Just create it, configure it, and wait for it to start. If any exceptions happen, they are handled the traditional way using available language constructions.
 
+== Server Lifecycle
+
+In Helidon 3.x you provided code to run after WebServer startup and after WebServer shutdown using asynchronous constructs:
```

Comment:
Is there an existing Helidon 3 doc page or example that shows this that the upgrade doc could link to?

It might be clearer if the upgrade doc did not include in-line the v3 code (especially not first) and showed only the v4 code.

Reply:
I followed the pattern that is used elsewhere in the migration guide -- showing the old v3 code and then the new v4 code.

Reply:
Shows what I know.